mantisshrimp/models/mantis_rcnn/mantis_mask_rcnn.py

```diff
@@ -5,20 +5,57 @@
 from mantisshrimp.models.mantis_rcnn.rcnn_param_groups import *
 from mantisshrimp.models.mantis_rcnn.mantis_rcnn import *
 from mantisshrimp.models.mantis_rcnn.mantis_faster_rcnn import *
+from mantisshrimp.backbones import *
 
 
 class MantisMaskRCNN(MantisRCNN):
     @delegates(MaskRCNN.__init__)
-    def __init__(self, n_class, h=256, pretrained=True, metrics=None, **kwargs):
+    def __init__(
+        self, n_class: int, backbone: nn.Module = None, metrics=None, **kwargs,
+    ):
         super().__init__(metrics=metrics)
-        self.n_class, self.h, self.pretrained = n_class, h, pretrained
-        self.m = maskrcnn_resnet50_fpn(pretrained=self.pretrained, **kwargs)
-        in_features = self.m.roi_heads.box_predictor.cls_score.in_features
-        self.m.roi_heads.box_predictor = FastRCNNPredictor(in_features, self.n_class)
-        in_features_mask = self.m.roi_heads.mask_predictor.conv5_mask.in_channels
-        self.m.roi_heads.mask_predictor = MaskRCNNPredictor(
-            in_features_mask, self.h, self.n_class
-        )
+        self.n_class = n_class
+        self.backbone = backbone
+
+        if backbone is None:
+            # Creates the default fasterrcnn as given in pytorch. Trained on COCO dataset
+            self.m = maskrcnn_resnet50_fpn(
+                pretrained=True, num_classes=n_class, **kwargs,
+            )
+            in_features = self.m.roi_heads.box_predictor.cls_score.in_features
+
+            self.m.roi_heads.box_predictor = FastRCNNPredictor(in_features, n_class)
+            in_features_mask = self.m.roi_heads.mask_predictor.conv5_mask.in_channels
+            hidden_layer = 256
+            self.m.roi_heads.mask_predictor = MaskRCNNPredictor(
+                in_features_mask, hidden_layer, self.n_class
+            )
+
+        else:
+            self.m = MaskRCNN(backbone, num_classes=n_class, **kwargs)
+
+    @staticmethod
+    def get_backbone_by_name(
```

Comment:
Is this exactly the same as for FasterRCNN? If so it can go to the parent class

Reply:
Yes it is exactly the same

Reply:
But it would look a bit odd this is code for MantisFasterRCNN.. And it would be same even for Mask RCNN so user would feel a bit odd.

```
mobibenet_v2_backbone = MantisFasterRCNN.get_backbone_by_name("mobilenet", fpn=False, pretrained=True)
```

Reply:
Sorry, I did not quite understood your point. Even if we move the code to the parent class the user would still be able to do exactly that

Reply:
So, you mean code be moved to MantisRCNN ? Since we inherit in both cases from that.

Reply:
Okay, sorry for the confusion. If we move to parent class, yes Indeed we can do it.